Source/dom/models/Document.js

```diff
@@ -686,6 +687,59 @@ Document.define('gradients', {
   },
 })
 
+/**
+ * A list of document swatches
+ *
+ * @return {Array<Swatch>} A mutable array of swatches defined in the document
+ */
+Document.define('swatches', {
+  array: true,
+  get() {
+    if (!this._object) {
+      return []
+    }
+    const documentData = this._getMSDocumentData()
+    return toArray(documentData.sharedSwatches().swatches()).map((a) =>
+      Swatch.fromNative(a)
+    )
+  },
+  set(swatches) {
+    if (this.isImmutable()) {
+      return
+    }
+    const assets = this._getMSDocumentData().sharedSwatches()
+    assets.removeSharedObjects(assets.swatches())
+    toArray(swatches)
+      .map((c) => Swatch.from(c))
+      .forEach((c) => {
+        const color = Color.from(c.color).toMSColor()
+        assets.addSwatchWithName_color(c.name, color)
+      })
+  },
+  insertItem(swatch, index) {
+    if (this.isImmutable()) {
+      return
+    }
+    const assets = this._getMSDocumentData().sharedSwatches()
+    const wrapped = Swatch.from(swatch)
+    const color = Color.from(wrapped.color).toMSColor()
+    if (index != undefined) {
+      // TODO: do something to insert the swatch at the right position
+      assets.addSwatchWithName_color_(wrapped.name, color)
+    } else {
+      assets.addSwatchWithName_color_(wrapped.name, color)
+    }
+  },
+  removeItem(index){
+    if (this.isImmutable()) {
+      return
+    }
+    const assets = this._getMSDocumentData().sharedSwatches()
+    const swatch = assets.swatchAt(index)
+    assets.removeSharedObjects([swatch])
+  },
```

Comment:
If swatches are internally an unordered sequence it may be better to implement it as a set instead of an array 🤔

Reply:
AFAIK they are ordered (or so I wrote on my dev notes: "Internally, swatches are returned in order, and apparently the order does not change when changing a Swatch name")